examples/denoising.jl

```diff
@@ -20,11 +20,11 @@ imgn= img .+ .01f0*randn(Float32, size(img))
 b = A*vec(imgn)
 
 # setup bregamn
-opt = bregman_options(maxIter=200, verbose=2, quantile=.5, alpha=1, antichatter=true)
-opt2 = bregman_options(maxIter=200, verbose=2, quantile=.5, alpha=1, antichatter=true, spg=true)
+opt = bregman_options(maxIter=200, verbose=2, alpha=1, antichatter=true)
```

Comment:
I really don't like this. Now there is options all over the place that completely defeat the purpose. And why is the TD moved

Reply:
I remove TD so that it's optional (default is identity matrix)

What do you mean "Now there is options all over the place that completely defeat the purpose"?

Reply:
like do you mean I should move these lambda, lambdafunc etc inside the bregmanparams?

Reply:
Now there is the options and the additional kwargs. Like user have to set options then have to put new ones separately like lambda. I still think its a bit messy. Put back the old basic interface and use dispatch to properly set the other cases in a clean way not adding 50 kwargs

Reply:
Can't dispatch over a quantile or a pre-set threshold. They are both a single number. Any thought?

Reply:
99% of what you added can be easily done via simple dispatch rather than all these additional keyword arguments on top of the options, for exampel would have takien you one line to define

```
bregman(A, x, b) = bregman(A, LinearAlgebra.I, x::Array{T}, b)
```

And everything satays clean an easy to use.  Try to make an effort making it user friendly not "works for me" friendly. All the rest is the same can be easily done via dispatch and/or options preprocessing and setup.

Reply:
ok thanks for your suggestion on this `LinearAlgebra.I` thing. I agree this works in a more reasonable structure.

What's your take on this threshold thing? Following your suggestion, I should add it in the bregmanparams, right? Since a pre-set threshold and a quantile are both single value. Can't dispatch.

Reply:
Options are all kwargs so you can filter then and do so pre-setup based on input